core-bundle/tests/Contao/DataContainerTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Contao;
+
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\DC_Table;
+
+class DataContainerTest extends TestCase
+{
+    /**
+     * @dataProvider getCombinerValues
+     */
+    public function testCombiner(array $source, array $expected): void
+    {
+        $reflection = new \ReflectionClass(DC_Table::class);
+        $names = $reflection->getMethod('combiner')->invoke($reflection->newInstanceWithoutConstructor(), $source);
+
+        $this->assertSame($expected, array_values($names));
+    }
+
+    public function getCombinerValues(): array
+    {
+        return [
+            [
+                ['foo'],
+                ['foo'],
+            ],
+            [
+                ['foo', 'bar'],
+                ['bar', 'foobar', 'foo'],
+            ],
+            [
+                ['foo', 'bar', 'baz'],
+                ['baz', 'barbaz', 'bar', 'foobar', 'foobarbaz', 'foobaz', 'foo'],
```

Comment:
Also, this order seems totally weird to me and I’d like to change it to:

```php
['foobarbaz', 'barbaz', 'foobaz', 'foobar', 'baz', 'bar', 'foo']
```

But I’m not sure if this could cause BC breaks.

Reply:
I will leave this PR as it is then, and create a separate issue for changing this order.